packages/data-context/test/unit/codegen/spec-options.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { expect } from 'chai'
+import fs from 'fs-extra'
+import path from 'path'
+import { DataContext } from '../../../src'
+import { SpecOptions, expectedSpecExtensions } from '../../../src/codegen/spec-options'
+import { createTestDataContext } from '../helper'
+
+const tmpPath = path.join(__dirname, 'tmp/test-code-gen')
+
+describe('spec-options', () => {
+  let ctx: DataContext
+
+  beforeEach(async () => {
+    await fs.remove(tmpPath)
+
+    ctx = createTestDataContext()
+
+    ctx.update((s) => {
+      s.currentProject = tmpPath
+    })
+  })
+
+  describe('getCodeGenOptions', () => {
+    context('unique file names', () => {
+      for (const specExtension of expectedSpecExtensions) {
```

Comment:
nit: I like to add an additional assertion that the expected tests run when generating the suite like this, since otherwise a regression in `expectedSpecExtensions` could possibly not cause a failure. Even something like `expect(expectedSpecExtensions).to.deep.eq(['.cy', '.spec', '.test', '-spec', '-test', '_spec'])`. But I guess this could also be construed as testing the implementation, not the feature, so it's just a thought.

Reply:
Agreed, added a test to validate the array is as we expect.